playwright/masonry/scroll-performance.spec.mjs

```diff
@@ -0,0 +1,122 @@
+// @flow strict
+import fs from 'fs';
+import { expect, test } from '@playwright/test';
+import getServerURL from './utils/getServerURL.mjs';
+import waitForRenderedItems from './utils/waitForRenderedItems.mjs';
+
+const DEFAULT_SCROLL_COUNT = 5;
+const PINS_COUNT = 20;
+const PAGE_HEIGHT = 1000;
+
+const eventsToCsv = (
+  events /*: $ReadOnlyArray<{ name: string, fps?: number, ts: number }> */
+) =>
+  events
+    .map((e) => (e.name.includes('clock_sync') ? e.name : e.fps))
+    .join('\n');
+
+const getFpsFromEvents = (events) => {
+  const fpsTimings = [];
+
+  const drawFrameEvents = events
+    .filter((e) => e.name.includes('DrawFrame'))
+    .sort((a, b) => a.ts - b.ts);
+  const clockSyncEvents = events.filter((e) => e.name.includes('clock_sync'));
+
+  for (let i = 0; i < drawFrameEvents.length - 1; i += 1) {
+    const delta = (drawFrameEvents[i + 1].ts - drawFrameEvents[i].ts) / 10e5;
+    fpsTimings.push({
+      name: 'fps',
+      fps: 1 / delta,
+      ts: drawFrameEvents[i + 1].ts,
+    });
+  }
+
+  return [...fpsTimings, ...clockSyncEvents].sort((a, b) => a.ts - b.ts);
+};
+
+const getFpsAvgPerScroll = (events) => {
+  const fpsAvgPerScroll = [];
+  let fpsSum = 0;
+  let count = 0;
+
+  for (let i = 0; i < events.length; i += 1) {
+    if (events[i].name.includes('clock_sync')) {
+      fpsAvgPerScroll.push(fpsSum / count);
```

Comment:
ohh clever. so instead of starting/stopping the tracing, we're just marking the scroll events and using that to indicate when we should reset our fps calculations?

Reply:
Yes, I tried stopping/starting the tracing but I would have to wait each time for all the events to finish and this can make the test behave in strange ways, I find this approach better